rust/lance-index/src/scalar/btree.rs

```diff
@@ -50,6 +51,58 @@ const BTREE_PAGES_NAME: &str = "page_data.lance";
 #[derive(Clone, Debug)]
 struct OrderableScalarValue(ScalarValue);
 
+impl DeepSizeOf for OrderableScalarValue {
+    fn deep_size_of_children(&self, context: &mut deepsize::Context) -> usize {
+        match &self.0 {
+            ScalarValue::Null => 0,
+            ScalarValue::Boolean(_) => 0,
+            ScalarValue::Float32(_) => 0,
+            ScalarValue::Float64(_) => 0,
+            ScalarValue::Decimal128(_, _, _) => 0,
+            ScalarValue::Decimal256(_, _, _) => 0,
+            ScalarValue::Int8(_) => 0,
+            ScalarValue::Int16(_) => 0,
+            ScalarValue::Int32(_) => 0,
```

Comment:
Why make this zero? Also, could we use `ScalarValue::size()`? https://docs.rs/datafusion/latest/datafusion/common/enum.ScalarValue.html#method.size

Reply:
Can't believe I missed that :facepalm: .  Yes, that will work.  It's `0` because `deepsize` does the std::mem::size_of already so this is only supposed to account for anything above and beyond that.